rmw_fastrtps_dynamic_cpp/src/rmw_init.cpp

```diff
@@ -12,13 +12,88 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
+#include "rmw/impl/cpp/macros.hpp"
 #include "rmw/rmw.h"
 
+#include "rmw_fastrtps_dynamic_cpp/identifier.hpp"
+
 extern "C"
 {
 rmw_ret_t
-rmw_init()
+rmw_init_options_init(rmw_init_options_t * init_options, rcutils_allocator_t allocator)
+{
+  RMW_CHECK_ARGUMENT_FOR_NULL(init_options, RMW_RET_INVALID_ARGUMENT);
+  RCUTILS_CHECK_ALLOCATOR(&allocator, return RMW_RET_INVALID_ARGUMENT);
+  if (NULL != init_options->implementation_identifier) {
+    RMW_SET_ERROR_MSG("expected zero-initialized init_options");
+    return RMW_RET_INVALID_ARGUMENT;
+  }
+  init_options->instance_id = 0;
+  init_options->implementation_identifier = eprosima_fastrtps_identifier;
```

Comment:
Should there be an `eprosima_fastrtps_dynamic_identifier` like connext has?

Reply:
Maybe, but I think that would be follow up to the fastrtps dynamic pr's and not in scope for this pr.